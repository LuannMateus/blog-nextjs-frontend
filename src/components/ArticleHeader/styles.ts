import styled, { css } from 'styled-components';
import { Title as HeadingStyle } from '../Heading/styles';

export const Wrapper = styled.section`
  ${({ theme }) => css`
    padding-bottom: ${theme.spacings.xlarge};
    margin-bottom: ${theme.spacings.xlarge};
    border-bottom: 0.1rem solid ${theme.colors.mediumGray};

    ${HeadingStyle} {
      margin: 0;
      margin-bottom: ${theme.spacings.medium};
    }
  `}
`;

export const Excerpt = styled.p`
  ${({ theme }) => css`
    font-size: ${theme.font.sizes.medium};
    margin: ${theme.spacings.medium} 0;
  `}
`;

export const Cover = styled.img`
  ${({ theme }) => css`
    display: block;
    max-width: 100%;
    margin-bottom: ${theme.spacings.medium};
  `}
`;
