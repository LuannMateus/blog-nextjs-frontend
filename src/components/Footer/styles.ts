import styled, { css } from 'styled-components';

export const Container = styled.footer`
  ${({ theme }) => css`
    width: 100%;

    text-align: center;

    a {
      color: inherit;
      text-decoration: none;
      font-size: ${theme.font.sizes.small};
    }
  `}
`;
