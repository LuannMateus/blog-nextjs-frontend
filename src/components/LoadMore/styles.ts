import styled, { css } from 'styled-components';

export const Wrapper = styled.button`
  ${({ theme }) => css`
    width: 16rem;

    border: 1px solid ${theme.colors.darkText};

    background: ${theme.colors.primary};
    color: ${theme.colors.white};

    display: flex;
    justify-content: center;
    align-items: center;

    padding: ${theme.spacings.small};

    cursor: pointer;

    transition: filter 300ms ease-in-out;

    &:hover {
      filter: brightness(50%);
    }

    &:disabled {
      opacity: 0.4;
    }
  `}
`;
