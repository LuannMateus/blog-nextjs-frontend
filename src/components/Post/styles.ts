import styled, { css } from 'styled-components';

export const Wrapper = styled.section`
  ${({ theme }) => css`
    width: 100%;

    padding-left: ${theme.spacings.small};
  `}
`;
