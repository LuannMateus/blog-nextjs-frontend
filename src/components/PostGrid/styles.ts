import styled, { css } from 'styled-components';

export const Wrapper = styled.main`
  ${({ theme }) => css`
    max-width: 100%;
    width: 100%;

    margin: 0 auto;

    display: grid;
    grid-template-columns: repeat(auto-fill, minmax(28rem, 1fr));
    gap: ${theme.spacings.large};

    padding: ${theme.spacings.large};

    @media ${theme.media.lteSmall} {
      grid-template-columns: 1fr;
    }
  `}
`;

export const NotFound = styled.p`
  ${({ theme }) => css`
    text-align: center;
    font-size: ${theme.font.sizes.small};
    padding: ${theme.spacings.large};

    @media ${theme.media.lteSmall} {
      padding: 0;
    }
  `}
`;
