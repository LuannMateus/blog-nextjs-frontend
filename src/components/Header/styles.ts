import styled, { css } from 'styled-components';
import { Title as HeadingStyles } from '../Heading/styles';

export const Wrapper = styled.header`
  ${({ theme }) => css`
    width: 100%;

    display: flex;
    justify-content: center;
    align-items: center;

    padding-bottom: ${theme.spacings.xhuge};
    max-width: ${theme.sizes.max};
    color: ${theme.colors.darkText};
    font-size: ${theme.font.sizes.small};
    margin: 0 auto;

    ${HeadingStyles} {
      margin-right: calc(${theme.spacings.small} - 1rem);
    }

    @media ${theme.media.lteMedium} {
      margin: 0;

      flex-direction: column;
    }
  `}
`;

export const Content = styled.section`
  ${({ theme }) => css`
    display: flex;
    flex-flow: column wrap;
    margin-left: ${theme.spacings.medium};

    @media ${theme.media.lteMedium} {
      margin-top: ${theme.spacings.small};
      margin-left: ${theme.spacings.small};
    }
  `}
`;
