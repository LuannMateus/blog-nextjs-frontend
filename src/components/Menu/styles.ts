import styled, { css } from 'styled-components';
import { Container as LogoLinkStyles } from '../LogoLink/styles';

export const Wrapper = styled.nav`
  ${({ theme }) => css`
    width: 27rem;
    height: 100vh;

    display: flex;
    flex-direction: column;
    align-items: center;
    background-color: #000;
    border-radius: 4px;

    padding-top: 1.4rem;

    position: fixed;
    left: -26rem;

    transition: all 300ms ease-in-out;

    z-index: 1000;

    .menuIcon {
      position: relative;
      left: 1.5rem;

      background: #000;
      padding: 0;
    }

    svg {
      width: 3rem;
      color: ${theme.colors.secondary};

      align-self: flex-end;

      padding-right: 1.6rem;

      cursor: pointer;
      transition: filter 300ms ease-in-out;
    }

    svg:hover {
      filter: brightness(50%);
    }

    ${LogoLinkStyles} {
      margin-top: ${theme.spacings.xhuge};
    }

    &.active {
      left: 0rem;
    }
  `}
`;

export const MenuLinkContainer = styled.section`
  ${() => css`
    width: 90%;

    margin-top: 4rem;
  `}
`;
