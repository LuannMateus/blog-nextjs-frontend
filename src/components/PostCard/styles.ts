import styled, { css } from 'styled-components';
import { Title as HeadingStyles } from '../Heading/styles';

export const Wrapper = styled.section`
  ${({ theme }) => css`
    ${HeadingStyles} {
      margin: 0;
      margin-top: calc(${theme.spacings.small} - 0.6rem);
      margin-bottom: ${theme.spacings.small};
    }

    a {
      text-decoration: none;
      color: inherit;
      transition: all 300ms ease-in-out;
    }

    img {
      transition: filter 300ms ease-in-out;
      width: 100%;
      max-width: 100%;
    }

    &:hover a {
      color: ${theme.colors.secondary};
    }

    &:hover img {
      filter: brightness(80%);
    }
  `};
`;

export const Cover = styled.img`
  max-width: 100%;
  transition: opacity 300ms ease-in-out;
`;

export const Excerpt = styled.p``;
