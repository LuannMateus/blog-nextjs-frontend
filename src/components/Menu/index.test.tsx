import { fireEvent, screen } from '@testing-library/react';
import { renderTheme } from '../../styles/renderTheme';
import { Menu } from '.';

import props from './mock';

describe('<Menu />', () => {
  it('Should render OpenMenu button', () => {
    renderTheme(<Menu {...props} />);

    expect(screen.getByLabelText('Open menu')).toBeInTheDocument();
  });

  it('Should render Menu with links and logoLink', () => {
    renderTheme(<Menu {...props} />);

    const openMenuButton = screen.getByLabelText('Open menu');

    fireEvent.click(openMenuButton);

    expect(screen.getByRole('navigation')).toBeInTheDocument();
    expect(
      screen.getByRole('navigation').querySelectorAll('a:not([href="/"])'),
    ).toHaveLength(props.links.length);
    expect(screen.getByRole('img', { name: props.text })).toBeInTheDocument();
  });

  it('Should render Menu with no links', () => {
    renderTheme(<Menu {...props} links={undefined} />);

    expect(screen.queryByText('/LINK/gi')).not.toBeInTheDocument();
  });

  it('Should render Menu with open and close events', () => {
    renderTheme(<Menu {...props} />);

    expect(screen.queryByRole('navigation')).not.toBeInTheDocument();

    const openMenuButton = screen.getByLabelText('Open menu');
    fireEvent.click(openMenuButton);
    expect(screen.queryByRole('navigation')).toBeInTheDocument();

    const closeMenuButton = screen.getByLabelText('Close menu');
    fireEvent.click(closeMenuButton);
    expect(screen.queryByRole('navigation')).not.toBeInTheDocument();
  });

  it('Should match snapshot', () => {
    const { container } = renderTheme(<Menu {...props} />);

    expect(container).toMatchSnapshot();
  });
});
