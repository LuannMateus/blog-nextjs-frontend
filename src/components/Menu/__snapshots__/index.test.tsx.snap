// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Menu /> Should match snapshot 1`] = `
.c4 {
  line-height: 1.2;
  color: #000000;
  font-size: 2.4rem;
  text-transform: uppercase;
}

.c6 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-text-decoration: none;
  text-decoration: none;
  color: inherit;
}

.c6 > img {
  width: 15rem;
  height: 15rem;
  border-radius: 50%;
}

.c8 {
  display: block;
  color: #FFFFFF;
  -webkit-text-decoration: none;
  text-decoration: none;
  margin-bottom: 1.6rem;
  font-size: 1.8rem;
  border-right: 0.5rem solid #000000;
  -webkit-transition: all 300ms ease-in-out;
  transition: all 300ms ease-in-out;
}

.c8:hover {
  border-right: 0.5rem solid #dc143c;
  color: #dc143c;
}

.c2 {
  display: inline-block;
  vertical-align: middle;
  overflow: hidden;
}

.c0 {
  width: 27rem;
  height: 100%;
  background: #000000;
  border-top-right-radius: 4px;
  padding: 0.8rem calc(0.8rem - 0.6rem);
  position: fixed;
  left: -26rem;
  -webkit-transition: all 300ms ease-in-out;
  transition: all 300ms ease-in-out;
  z-index: 1001;
}

.c0 .c5 {
  margin-top: 6.4rem;
}

.c0 .c5 img {
  border: 0.5rem solid #dc143c;
}

.c0.active {
  left: 0rem;
  overflow-y: auto;
  overflow-x: hidden;
}

.c3 {
  width: 100%;
  max-width: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c1 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: end;
  -webkit-justify-content: flex-end;
  -ms-flex-pack: end;
  justify-content: flex-end;
  background-color: #000000;
  border-radius: 4px;
  left: 2rem;
}

.c1 svg {
  width: 3rem;
  color: #dc143c;
  padding-right: 3.5rem;
  padding-top: 1rem;
  cursor: pointer;
  -webkit-transition: filter 300ms ease-in-out;
  transition: filter 300ms ease-in-out;
}

.c1 svg:hover {
  -webkit-filter: brightness(50%);
  filter: brightness(50%);
}

.c1 .menuIcon {
  padding: 0;
}

.c7 {
  width: 90%;
  margin: 4rem 0;
}

@media (max-width:768px) {
  .c0 {
    width: 25.5rem;
  }
}

<div>
  <div
    class="c0"
  >
    <div
      class="c1"
    >
      <svg
        aria-label="Open menu"
        class="c2 menuIcon"
        fill="currentColor"
        focusable="false"
        role="img"
        viewBox="0 0 24 24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <title>
          Open menu
        </title>
        <path
          d="M0 0h24v24H0V0z"
          fill="none"
        />
        <path
          d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
        />
      </svg>
    </div>
    <nav
      aria-hidden="true"
      class="c3"
    >
      <h1
        class="c4"
      >
        <a
          class="c5 c6"
          href="/"
          target="_self"
        >
          <img
            alt="blog-name"
            src="logo-image"
          />
        </a>
      </h1>
      <section
        class="c7"
      >
        <a
          class="c8"
          href="https://www.google.com/"
          target="_self"
        >
          LINK 1
        </a>
        <a
          class="c8"
          href="https://www.google.com/"
          target="_self"
        >
          LINK 2
        </a>
        <a
          class="c8"
          href="https://www.google.com/"
          target="_self"
        >
          LINK 3
        </a>
        <a
          class="c8"
          href="https://www.google.com/"
          target="_self"
        >
          LINK 4
        </a>
      </section>
    </nav>
  </div>
</div>
`;
